src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -287,12 +289,32 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                 page = JsonParser.toJava(response, BitbucketServerPullRequests.class);
                 pullRequests.addAll(page.getValues());
             }
+            
+            for (BitbucketServerPullRequest pullRequest : pullRequests) {
+				pullRequest.setCanMerge(getPullRequestCanMergeById(Integer.parseInt(pullRequest.getId())));
```

Comment:
This is intentional. As described in JENKINS-45997. Bitbucket Server has the (unfortunate) behaviour that refs/pull-requests/3/from and refs/pull-requests/3/merge are only getting updated when someone visits the pull request website. The workaround is to call the merge api as this will update the refs as well.

So this pull request is to call the "can merge" api to make sure that the refs are always up to date.
This should fix JENKINS-45997 and hopefully also JENKINS-45973 and JENKINS-46174

The trait I plan to do will then be something that will build on top of this.

Reply:
Oh btw, I just noticed... Friends don't let friends use tabs 😄